src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/Completion.cs

```diff
@@ -232,9 +232,10 @@ private static TextEdit BuildRevertedEdit(TextEdit provisionalTextEdit)
     public async Task<CompletionItem?> ProvideResolvedCompletionItemAsync(DelegatedCompletionItemResolveParams request, CancellationToken cancellationToken)
     {
         // Check if we're completing a snippet item that we provided
-        if (SnippetCompletionData.TryParse(request.CompletionItem.Data, out var snippetCompletionData))
+        if (SnippetCompletionData.TryParse(request.CompletionItem.Data, out var snippetCompletionData) &&
+            _snippetCache.TryResolveSnippetString(snippetCompletionData) is { } snippetInsertText)
         {
-            request.CompletionItem.InsertText = _snippetCache.TryResolveSnippetString(snippetCompletionData);
+            request.CompletionItem.InsertText = snippetInsertText;
```

Comment:
Strictly speaking, with the fix in the `TryParse` method, this change is redundant, but it seems like a good idea since `TryResolveSnippetString` can happily return null, we should probably not clobber whatever data was there before.